### ISSUE TITLE
smtp: Failed: Remote host not set with latest HEAD

### ISSUE THREAD

**Issue by dev-zero** [bruceg/nullmailer#28]
with the latest version I get:

```
Rescanning queue.
Starting delivery, 1 message(s) in queue.
Starting delivery: protocol: smtp host: mail.my-hoster.com file: 1439464287.9944
smtp: Failed: Remote host not set
Sending failed: Unspecified temporary error
Delivery complete, 1 message(s) remain.
```

with the same configuration in ```remotes``` which works with 5e9316c669e752fcd9058b28952919bccd967640:

```
mail.my-hoster.com smtp --port=587 --starttls --auth-login --user=myuser@my-domain.com --pass=s3cr3t
```

**Comment by marcecj** [comment]
FWIW, 29734e270ab05f9493c4c8bce189ccda1e5669a5 is broken, too, so the bug must have been introduced somewhere between commits 5e9316c669e752fcd9058b28952919bccd967640 and 29734e270ab05f9493c4c8bce189ccda1e5669a5.  Ignoring the NEWS update and the merge commit leaves us with just 4 relevant commits.

**Comment by bruceg** [comment]
Fixed in ce2b78c
